src/components/Questionnaire/ValueSetSelect.tsx

```diff
@@ -65,12 +80,125 @@ export default function ValueSetSelect({
     }),
   });
 
+  const favouritesQuery = useQuery({
+    queryKey: ["valueset", system, "favourites"],
+    queryFn: query(valuesetRoutes.favourites, { pathParams: { slug: system } }),
+  });
+
+  const addFavouriteMutation = useMutation({
+    mutationFn: (body: { code: string }) =>
```

Comment:
Since we are specifying type in the API, should be able to directly call mutate here. Same for those below.

```suggestion
    mutationFn: mutate(valuesetRoutes.addFavourite, {
      pathParams: { slug: system },
    }),
```